pkg/util/arith.go

```diff
@@ -33,3 +33,31 @@ func Min[T constraints.Ordered](x, y T) T {
 		return y
 	}
 }
+
+// AtomicInt represents the shared interface provided by various atomic.<NAME> integers
+//
+// This interface type is primarily used by AtomicMax.
+type AtomicInt[I any] interface {
```

Comment:
What do you mean?

Reply:
I believe that this is valid: `name := AtomicInt[string]`, because `I` can be any type, not just an integer, which is confusing.

Reply:
`AtomicInt` is an interface, so you can't do that directly. You could 100% declare a type like that, but probably will run into trouble constructing it (because realistically nothing will have the appropriate methods returning `string`s)